### PR TITLE
feat: Add database routing support

### PR DESCRIPTION
**Pull request by jrconlin**
This adds support to route data to the appropriate storage backend.

Requires: #369 

Closes: SYNC-3452, SYNC-3961

### CODE REVIEW

File: autopush-common/src/db/dynamodb/mod.rs [hunk 1]
```diff
@@ -40,6 +40,12 @@ pub struct DynamoDbSettings {
     pub router_table: String,
     #[serde(default)]
     pub message_table: String,
+    #[serde(default)]
+    pub message_table_names: Vec<String>,
+    #[serde(default)]
+    pub current_message_month: String,
```
[the remark this answers is not in the document]
Comment (jrconlin) [reply autopush-common/src/db/dynamodb/mod.rs:46]: Ah, good catch. missed those.

File: Cargo.toml [hunk 2]
```diff
@@ -78,7 +78,9 @@ rusoto_dynamodb = { version = "0.47", default-features = false, features = [
 sentry = { version = "0.31", features = [
   "debug-logs",
 ] } # Using debug-logs avoids https://github.com/getsentry/sentry-rust/issues/237
-sentry-core = { version = "0.31" }
+sentry-core = { version = "0.31", default-features = false, features = [
+  "client",
+] }
```
Comment (pjenvey) [inline Cargo.toml:83]: I think this was in another PR as well -- we don't use this feature AFAICT

```suggestion
sentry-core = { version = "0.31" }
```

File: autopush-common/src/db/bigtable/bigtable_client/mod.rs [hunk 3]
```diff
@@ -141,6 +141,7 @@ impl BigTableClientImpl {
         &self,
         row_key: &str,
         timestamp_filter: Option<u64>,
+        table_name: Option<String>,
```
Comment (pjenvey) [inline autopush-common/src/db/bigtable/bigtable_client/mod.rs:144]: Why the new optional argument? I don't see it being used (`None` is always passed)

**Review by pjenvey — COMMENTED** [review]
sorry, missed a couple more things

**Comment by jrconlin** [comment]
Yeah, this looks like more merge detritus. (there was a lot.)

File: autopush-common/src/db/bigtable/bigtable_client/mod.rs [hunk 4]
```diff
@@ -178,15 +178,19 @@ impl BigTableClientImpl {
     /// write a given row.
     ///
     /// there's also `.mutate_rows` which I presume allows multiple.
-    async fn write_row(&self, row: row::Row) -> Result<(), error::BigTableError> {
+    async fn write_row(
+        &self,
+        row: row::Row,
+        table_name: Option<String>,
```
Comment (pjenvey) [inline autopush-common/src/db/bigtable/bigtable_client/mod.rs:184]: There's still this one for `write_row`

**Review by pjenvey — CHANGES_REQUESTED** [review]

**Review by pjenvey — APPROVED** [review]

**Pull request merged**